test/files/jvm/javaReflection/Test.scala

```diff
@@ -50,23 +50,30 @@ getSimpleName / getCanonicalName / isAnonymousClass / isLocalClass / isSynthetic
     will change some day).
 */
 
+import scala.tools.nsc.settings.ScalaVersion
+import scala.util.Properties.javaSpecVersion
+
 object Test {
 
+  def assert8(b: => Boolean, msg: => Any) = {
+    if (ScalaVersion(javaSpecVersion) == ScalaVersion("1.8")) assert(b, msg)
+    else if (!b) println(s"assert not $msg")
+  }
+
   def tr[T](m: => T): String = try {
     val r = m
     if (r == null) "null"
     else r.toString
   } catch { case e: InternalError => e.getMessage }
 
-  def assertNotAnonymous(c: Class[_]) = {
-    val an = try {
+  def assertNotAnonymous(c: Class[_]) = assert8(!isAnonymous(c), s"$c is anonymous")
```

Comment:
This check also flipped.

Reply:
IIUC, this is just documenting the evolving status quo for java reflect interaction.